File: proveedor-service/src/main/java/tingeso/microservicios/Services/proveedorService.java
package tingeso.microservicios.Services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import tingeso.microservicios.Entity.proveedorEntity;
import tingeso.microservicios.Repositories.proveedorRepository;

import java.util.List;

@Service
public class proveedorService {
    @Autowired
    proveedorRepository Proveedorrepository;

    public proveedorEntity save(proveedorEntity proveedor){
        proveedorEntity proveedorNew = Proveedorrepository.save(proveedor);
        return proveedorNew;
    }

    public List<proveedorEntity> getProveedores(){
        return Proveedorrepository.findAll();
    }

    public proveedorEntity getByCodigo(String codigo){
        proveedorEntity proveedorEntregar = new proveedorEntity();
        List<proveedorEntity> proveedores = getProveedores();
        for(int i = 0; i< proveedores.size(); i++){
            if(proveedores.get(i).getCodigo().equals(codigo)){
                proveedorEntregar = proveedores.get(i);
            }
        }
        return proveedorEntregar;
    }
}

File: milkstgo_frontend/src/services/ProveedorService.js
import axios from 'axios';

const PROVEEDOR_API_URL = "http://127.0.0.1:52141/proveedor";

class proveedorService {

    getProveedores(){
        return axios.get(PROVEEDOR_API_URL);
    }

    createProveedor(proveedor){
        return axios.post(PROVEEDOR_API_URL, proveedor);
    }

}

export default new proveedorService()